### PR TITLE
fix: create release action when there is no previous tag

### PR DESCRIPTION
**Pull request by ridwan-b**
If there is no previous tag, then the pipeline failed because of an undefined property.
This PR is to fix this.

### CODE REVIEW

File: create-release/index.js [hunk 1]
```diff
@@ -37,14 +37,16 @@ async function main() {
         const env = getEnvironment(tagSplitted[2]);
         previousTag = getPreviousTagWithEnv(allTags.data, env);
     }
+
+    let previousTagName = previousTag ? previousTag.name : "No previous tag";
```
Comment (cforycki) [inline create-release/index.js:41]:
```suggestion
    const previousTagName = previousTag ? previousTag.name : "No previous tag";
```

By the way, is this really working? 
I can't find mention of `"No previous tag"` in the [github api doc](https://docs.github.com/en/rest/releases/releases#generate-release-notes-content-for-a-release)

**Review by cforycki — COMMENTED** [review]
Which version of node did you use to compile?
Having that much diff in the generated code is a bit weird so I tried to compile it myself and I don't have the same output as you.

File: create-release/index.js (hunk 1 above)
Comment (ridwan-b) [reply create-release/index.js:41]: Actually, I thought it was a declarative field where I could put anything.
I should have read the doc before, it seems that if there is not previous tag, we have to let it empty. I'll change this

**Comment by ridwan-b** [comment]
This new version has been successfully tested on kleanad-prebid [](https://github.com/pbstck/kleanads-prebid/releases)

File: create-release/index.js [hunk 2]
```diff
@@ -37,15 +37,20 @@ async function main() {
         const env = getEnvironment(tagSplitted[2]);
         previousTag = getPreviousTagWithEnv(allTags.data, env);
     }
-    
-    console.log('Previous tag', previousTag.name)
-    console.log('tag',github.context.payload.ref)
-    const releaseNoteGenerated = await octokit.rest.repos.generateReleaseNotes({
+
+    let dataReleaseNoteGenerated = {
```
Comment (cforycki) [inline create-release/index.js:41]:
```suggestion
    const dataReleaseNoteGenerated = {
```

File: create-release/index.js [hunk 3]
```diff
@@ -37,15 +37,20 @@ async function main() {
...
         owner,
         repo,
         tag_name: github.context.payload.ref,
-        previous_tag_name: previousTag.name
-    })
+    };
+
+    if (previousTag) {
+        dataReleaseNoteGenerated.previous_tag_name = previousTag?.name;
+        console.log('Previous tag', previousTag?.name)
```
Comment (cforycki) [inline create-release/index.js:49]: No need for optional chaining here, you already check the variable in the if condition.
```suggestion
        dataReleaseNoteGenerated.previous_tag_name = previousTag.name;
        console.log('Previous tag', previousTag.name)
```

**Review by cforycki — CHANGES_REQUESTED** [review]

**Review by cforycki — APPROVED** [review]

**Review by slezier — APPROVED** [review]

**Review by lestrade-pbstck — APPROVED** [review]

**Review by ildede — APPROVED** [review]

**Review by vmaerten — APPROVED** [review]

**Review by afigoureux-pbstck — APPROVED** [review]

**Pull request merged**